packages/repository/src/model.ts

```diff
@@ -329,9 +330,9 @@ export abstract class Model {
 
     const json: AnyObject = {};
     const hiddenProperties: string[] = def.settings.hiddenProperties || [];
-    for (const p in def.properties) {
-      if (p in this && !hiddenProperties.includes(p)) {
-        copyPropertyAsJson(p);
+    for (const [pKey, pDef] of Object.entries(def.properties)) {
+      if (pKey in this && !hiddenProperties.includes(pKey) && !pDef.hidden) {
+        copyPropertyAsJson(pKey);
```

Comment:
Since you are not changing the behavior here, please revert this block back to original version to avoid code churn and make the future code archaeology easier.